### DISCUSSION TITLE
Do you have any plans to provide Compiling to Native Code?  Like UWP/.NetNative Does?

### DISCUSSION THREAD

**Discussion by groovykool** [microsoft/WindowsAppSDK#1631] [Q&A]
Do you have any plans to provide Compiling to Native Code?  Like UWP/.NetNative Does?

**Comment by jonwis** [comment]
WindowsAppSDK itself is native code, but has no opinion about the kind of code calling it.

Are you asking whether .NET will support pre-JIT so you can ship your own binaries as native instead of IL?

**Reply by groovykool** [reply]
> Are you asking whether .NET will support pre-JIT so you can ship your own binaries as native instead of IL?


Yes.  UWP applications are native code and startup and run faster than  IL applications.  Also IL applications can be very easily reverse engineered.

I would like to see  WindowsAppSDK compile to native code instead of IL.

**Reply by groovykool** [reply]
Microsoft on .NET Native [Reference](https://docs.microsoft.com/en-us/windows/uwp/dotnet-native/)
